### DISCUSSION TITLE
Why does it use so much memory (RAM) while running in the background?

### DISCUSSION THREAD

**Discussion by JusByte** [AutoDarkMode/Windows-Auto-Night-Mode#390] [Q&A]
I was just curious why there is a high memory usage (20-30MB) while idle in the background.  Thanks!

**Comment by Spiritreader** [comment]
Sorry to be a bit blunt, but in what world is 30MB even considered high RAM usage?
ADM is a dotnet 6 application and will pretty much use as much RAM as the dotnet assembly requires. 
The application programatically does not allocate any large RAM blocks, nor is it using many libraries.

When the program starts the entire dotnet assembly has to be loaded into RAM (which is still a lot smaller than an electron app). The Winforms framework used for the context menu also uses RAM when you click it.

However, the service automatically frees resources over time and will return to about 10MB
![grafik](https://user-images.githubusercontent.com/18438899/144328967-0b1e3b34-6a2d-40ce-b537-0b0dcbea283d.png)

**Reply by JusByte** [reply]
Ah ok.  I was just curious about the reason, and I do see the reduction to 10MB.  30MB is not high; I just meant high for when my computer was idle as it was higher than some core parts of main software while in the background.

Thanks for your quick reply, and I appreciate the work on this program!  Have a great day!

**Reply by Spiritreader** [reply]
One cool thing about the new dotnet framework is that it is technically platform independent. (But not very useful for Auto Dark Mode)
However that comes with the disadvantage that both the disk usage for shipping the app, as well as the run time memory usage is higher.
There are trimming options available, but getting this to work currently is sadly a nightmare.

The reason why ADM uses more RAM is pretty much because it's a modern C# application.
The updater for example, is written in rust and compiles to C, so it's both a smaller footprint on disk as in memory.
